### PR TITLE
lib/maths: add `nan` and `inf` as special Float

### PR DESCRIPTION
**Pull request by privat**
with bits of IEEE 754

### CODE REVIEW

**Comment by xymus** [comment]
+1

File: lib/standard/math.nit [hunk 1]
```diff
@@ -285,6 +293,12 @@ redef class Float
 	#  * 1 if self is positive infinity
 	#  * -1 if self is negative infinity
 	#  * 0 otherwise
+	#
+	# ~~~
+	# assert 10.0.is_inf == 0
+	# assert inf.is_inf == 1
+	# assert (-inf).is_inf == -1
+	# ~~~
 	fun is_inf: Int do
```
Comment (R4PaSs) [inline lib/standard/math.nit]: Now that I see this, was there any reason why this returns an `Int` and not a `Bool` ?

**Comment by R4PaSs** [comment]
:shipit:

File: lib/standard/math.nit (hunk 1 above)
Comment (Morriar) [inline lib/standard/math.nit]: It is used to differentiate positive and negative infinite

Comment (R4PaSs) [inline lib/standard/math.nit]: Good point there sir

**Comment by Morriar** [comment]
+1

**Pull request merged**